src/VirtoCommerce.NotificationsModule.Data/Model/NotificationEntity.cs

```diff
@@ -69,8 +69,7 @@ public virtual Notification ToModel(Notification notification)
             notification.IsActive = IsActive;
             notification.Type = Type;
 
-            notification.Templates = notification.Templates?.Concat(Templates
-                .Select(t => t.ToModel(AbstractTypeFactory<NotificationTemplate>.TryCreateInstance($"{Kind}Template")))).Distinct().ToList();
+            notification.Templates = MergeTemplates(notification);
```

Comment:
Group templates according to rules is a good candidate for the policy that can be moved and used on the business logic layer.